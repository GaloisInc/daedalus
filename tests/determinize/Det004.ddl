-- testing LL(k) with calling grammar some semantic value recomposition

def Main = { x = Gram1 ; y = Gram2, z = Gram3 }

def Gram1 =
     { Match "a"; Gram2}
  <| { Match "a"; Gram3}

def Gram2 =
   $['b']

def Gram3 =
  $['c']

