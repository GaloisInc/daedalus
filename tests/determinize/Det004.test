Det004.ddl
--dump-core
--determinize
