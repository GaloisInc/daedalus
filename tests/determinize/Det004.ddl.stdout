module Det004
 
--- Imports:
 
--- Type defs:
type Det004.Main = { x: uint 8
                   ; y: uint 8
                   ; z: uint 8
                   }
 
--- Rules:
 
Det004.Gram2 : Grammar (uint 8) =
  Match {'b'}
 
Det004.Gram3 : Grammar (uint 8) =
  Match {'c'}
 
Det004.Gram1 : Grammar (uint 8) =
  Choose biased
    { do @MatchBytes "a"
         ($$ : uint 8) <- Det004.Gram2
         pure $$
    | do @MatchBytes "a"
         ($$ : uint 8) <- Det004.Gram3
         pure $$
    }
 
Det004.Main : Grammar Det004.Main =
  do (x : uint 8) <- Det004.Gram1
     (y : uint 8) <- Det004.Gram2
     (z : uint 8) <- Det004.Gram3
     pure {x = x,
           y = y,
           z = z}
 
Det004.Gram2_9 : Grammar {} =
  @Match {'b'}
 
Det004.Gram3_10 : Grammar {} =
  @Match {'c'}
 
Det004.Gram1_11 : Grammar {} =
  Choose biased
    { do @MatchBytes "a"
         Det004.Gram2_9
    | do @MatchBytes "a"
         Det004.Gram3_10
    }
 
Det004.Main_12 : Grammar {} =
  do Det004.Gram1_11
     Det004.Gram2_9
     Det004.Gram3_10
